controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -602,6 +613,446 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         return result;
     }
 
+    /**
+     * This method sends a WireCommand to create a table segment.
+     *
+     * @param scope Stream scope.
+     * @param stream Stream name.
+     * @param segmentId Id of the segment to be created.
+     * @param hostControllerStore Host controller store.
+     * @param clientCF Client connection factory.
+     * @param delegationToken The token to be presented to the segmentstore.
+     * @param clientRequestId Request id.
+     * @return A CompletableFuture that, when completed normally, will indicate the table segment creation completed successfully.
+     * If the operation failed, the future will be failed with the causing exception. If the exception is retryable then the future
+     * will be failed with {@link WireCommandFailedException}.
+     */
+    public CompletableFuture<Boolean> createTableSegment(final String scope,
+                                                         final String stream,
+                                                         final long segmentId,
```

Comment:
My interest here is that we should be able to create multiple table segments for storing stream's metadata. Each corresponding to a separate class of records. 
So i will generate the stream name to reflect the type of table. 
So i dont necessarily need a segment number to go along with it. So we can remove that from the signature here.

Reply:
done, refactored all the apis.